### ISSUE TITLE
Long sentences causes minecraft crash

### ISSUE THREAD

**Issue by jho5245** [cubex2/BetterFonts#20]
When I type a long sentence that is more than 100 characters on chat, the game crashes!
Even other people's long chat sentence also crashes minecraft.
It happens also when I type a long command on the command block.
I'm using antiAlias and dropShadow to true, and the font is Comic Sans MS, font size is 18.

**Comment by CatMe0w** [comment]
Same issue.

**Comment by cubex2** [comment]
What minecraft and mod version are you using? Can you provide a crash report?

**Comment by jho5245** [comment]
Minecraft is 1.12.1 and the mod version is TTFR-1.12.1-1.9.1, which is released Aug 13, 2017
Forge version is forge1.12.1-14.22.0.2467 and I'm using LiteLoader and optifine mod.
In singleplayer it seems not to crash minecraft. but when in multiplayer, the game stops!
in fact, this is no crash, just stops. May be infinite loop of something exists I think. because no error log is 
left.

Anyway, this is the latest log!
https://pastebin.com/CLvqjNV2

**Comment by cubex2** [comment]
Can you remove optifine and check if it still stops working?

**Comment by jho5245** [comment]
I had already tried it both with and without using optifine. but the same result.

**Comment by cubex2** [comment]
I'm not able to reproduce this. Are you using any other mods?

**Comment by jho5245** [comment]
Nope, I only used TTFR mod.

**Comment by jho5245** [comment]
When I type long sentence in chat, but it is not that long so that the message doesn't change the chatline, minecraft doesn't stop, but when I chat too long messsage so that the message change the chatline, minecraft stops.

I also used various fonts to check it out, the bigger font, the easier the message changes the chatline.

**Comment by jho5245** [comment]
I'll try it again after deleting the whole minecraft data file.

**Comment by jho5245** [comment]
sadly, the same result, I deleted all the minecraft files and only used forge and TTFR with default font settings, default minecraft settings. The game still stops! :(

**Comment by cubex2** [comment]
Can you provide a sentence that makes the game stop?

**Comment by jho5245** [comment]
any message that doesn't have space bar (Word spacing)

**Comment by jho5245** [comment]
Here's a video
sorry for poor resolution
https://www.youtube.com/watch?v=FevgznWSNRE&feature=youtu.be

**Comment by cubex2** [comment]
That video doesn't work for me. It says the video is not available.

**Comment by jho5245** [comment]
Now is available.

**Comment by jho5245** [comment]
Is this can be fixed?

**Comment by cubex2** [comment]
I still can't reproduce it, so there currently is no way for me to fix it.

**Comment by jho5245** [comment]
so sad :(
I think this is kind of something when a player types a long sentence on sign, the end of some words are not rendered when he's using big size font (about 18 size)
maybe becuase this error might have been happened since that feature is modified.

Below is an example when I typed long messages with 12 size of sanserif, then change the size to 18.
And some words are not visible after changing the font size.
![capture](https://user-images.githubusercontent.com/11700801/30158669-f3247a5e-9400-11e7-83e8-115a576911ea.PNG)

And 18 size font
![capture2](https://user-images.githubusercontent.com/11700801/30158690-02b10d48-9401-11e7-87ad-ef51184f72e6.PNG)

**Comment by jho5245** [comment]
Maybe that long sentences causes infinity looping for some reasons, so as to stop minecraft.
Because at that moment, the cpu's usage icnreases up to almost 100% due to the infinity loop.
Kind of rendering issue?

**Comment by jho5245** [comment]
Can you please fix this? I really like this mod but the only glitch makes me don't want to use this mod :(

**Comment by jho5245** [comment]
Here's the last one I report the long chat message causes minecraft stop.

 

https://www.youtube.com/watch?v=QrljdDu5XsY&amp;feature=youtu.be

 

I'm really really sorry for keep saying this bug. But I also really hope that this bug is fixed.

 

Tried in only that mod and server. In singleplayer, this bug somehow doesn't crash minecraft.

**Comment by cubex2** [comment]
Can you test this version and tell me if it's working?
https://www.mediafire.com/file/heqrjut2rhu6d45/TTFR-1.11.2-1.7.2.jar

**Comment by jho5245** [comment]
Yes, this causes the same result, I've test only TTFR mod and the latest 1.11.2 forge version.

Edit:
and I hope you check this mod. This mod doesn't make minecraft crash when long chat is typed.
And has awesome features but has less compatibility with optifine :(
https://minecraft.curseforge.com/projects/unicode-font-extension?gameCategorySlug=mc-mods&projectID=279563

And when using both TTFR and that mod, the rendering is kinda buggy, but no crash when long chat is typed!

**Comment by eslym** [comment]
i found out when there is large update on string which need to be rendered, the UI thread will stuck, it causing the minecraft stop responding, mostly happen when long command used or typed in command block

**Comment by IceSelkie** [comment]
I can recreate this crash 100% of the time.

Minecraft freezes, then if you let it sit at 100% cpu for like 10 minutes it crashes on "ticking entity" for reason of using entire heapspace.

Forge: 1.12.2-14.23.4.2705
TTFR: 1.12.1-1.9.1

Ways to crash:
A chest full of items dense with nbt: (ex: slightly used diamond chestplates, fully enchanted, and named some really really long name), and fill the entire chest with it. Stand on top and run the command "/blockdata ~ ~-1 ~ {}", which should tell you the nbt data of the block you are standing on, but it is too long, and it freezes then crashes.
If that doesn't crash, it should at least freeze for a little bit. ctrl + pickblock to pick up the chest with nbt, and fill another chest with those, and run the command on that chest. That'll crash.
